domain/application/service/unit.go

```diff
@@ -110,6 +111,10 @@ type UnitState interface {
 	// If the unit is dead, an error satisfying [applicationerrors.UnitIsDead]
 	// is returned.
 	SetUnitConstraints(ctx context.Context, inUnitUUID coreunit.UUID, cons constraints.Constraints) error
+
+	// SetRunningAgentBinaryVersion sets the running agent version for the unit.
+	// A [applicationerrors.UnitNotFound] error will be returned if the unit does not exist.
```

Comment:
The `core errors.NotSupported` error is raised by this implementation and should be documented here as well I feel.